### ISSUE TITLE
update README links to tagged releases on pypi and npm

### ISSUE THREAD

**Issue by kristapratico** [Azure/azure-sdk-tools#3801]
Links that point to samples, etc in the repo that are found on PyPi and npm pages are not updated to the tagged release links.

Examples (click on "Samples" for either page and they go to main):

https://pypi.org/project/azure-ai-textanalytics/
https://www.npmjs.com/package/@azure/ai-text-analytics

If we can update these links to the tagged release versions this would be helpful for customers. We already do this for github.io and MS docs so hopefully that solution can just be applied here?

**Comment by scbedd** [comment]
Discussed with @danieljurek 

What we came up with

- [ ] Roll new script `eng/common/Insert-Release-Links.ps1`  -artifacts "${{ convertToJson(parameters.listOfValues) }}" -packageInfo <packageInfoDirectory>`
  - `artifacts` should be the list of yml artifacts that we normally pass into the `build` job from `ci.yml`. We will use these for targeting
  - Start with only README.md and CHANGELOG.md for link replacement
  - Pull the target version from PackageInfo.json for each package. Pulling from this packageinfo will save us needing to do language-specific digging interactions with `Language-Settings.ps1`
- [ ] Insert new script into `build` stage after PackageInfo is generated and _prior_ to build. We need the **files being packaged to have the replaced links**.
  - This replacement should _only_ run on `manual` and `internal` builds. It should be disable-able by a queue time build variable if we absolutely need it to be. Let's call that. `DISABLE_RELEASE_LINK_REPLACEMENT`

**Comment by scbedd** [comment]
I don't believe we currently do link replacement on ANY of our package managed items. @benbp @weshaggard for priority on our action on this.

**Comment by weshaggard** [comment]
@scbedd is correct we generally avoid updating any files only in packages to be different then when they are in the source code. I know @johanste has also expressed similar concerns with this in the past because we want the packages to be identical whether installed from pypi or the repo.
